core/src/main/java/overflowdb/NodeDb.java

```diff
@@ -421,6 +421,26 @@ public void addEdgeSilent(String label, Node inNode, Map<String, Object> keyValu
     return multiIterator;
   }
 
+  protected int outEdgeCount() {
+    int outEdgeCount = 0;
+    for (String label : layoutInformation().allowedOutEdgeLabels()) {
+      int offsetPos = getPositionInEdgeOffsets(Direction.OUT, label);
+      if (offsetPos != -1) {
+        int start = startIndex(offsetPos);
+        int length = blockLength(offsetPos);
+        int strideSize = getStrideSize(label);
+        for (int i = start;
+             i < adjacentNodesWithProperties.length && i < start + length;
```

Comment:
compute `start + length` and `adjacentNodesWithProperties.length` only once?

Reply:
`adjacentNodesWithProperties` is an array, so `.length` doesn't compute anything, and JIT will likely even inline it

`start + length`: yup ok will do across the board in this file